centreon/packages/ui/src/components/Form/Dashboard/DashboardForm.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { ReactElement, useCallback, useMemo } from 'react';
+
+import * as Yup from 'yup';
+import { useTranslation } from 'react-i18next';
+
+import { InputType } from '../../../Form/Inputs/models';
+import { Form, FormProps } from '../../../Form';
+import { FormVariant } from '../Form.models';
+
+import { useStyles } from './DashboardForm.styles';
+import {
+  labelCharacters,
+  labelMustBeAtLeast,
+  labelMustBeMost,
+  labelRequired
+} from './translatedLabels';
+import {
+  DashboardFormActions,
+  DashboardFormActionsProps
+} from './DashboardFormActions';
+import { DashboardResource } from './Dashboard.resource';
+
+export type DashboardFormProps = {
+  labels: DashboardFormLabels;
+  onSubmit?: FormProps<DashboardResource>['submit'];
+  resource?: DashboardResource;
+  variant?: FormVariant;
+} & Pick<DashboardFormActionsProps, 'onCancel'>;
+
+export type DashboardFormLabels = {
+  actions: DashboardFormActionsProps['labels'];
+  entity: Required<DashboardResource>;
+};
+
+const DashboardForm = ({
+  variant = 'create',
+  resource,
+  labels,
+  onSubmit,
+  onCancel
+}: DashboardFormProps): ReactElement => {
+  const { classes } = useStyles();
+  const { t } = useTranslation();
+
+  const formProps = useMemo<FormProps<DashboardResource>>(
+    () => ({
+      initialValues: resource ?? { name: '' },
+      inputs: [
+        {
+          fieldName: 'name',
+          group: 'main',
+          label: labels?.entity?.name,
+          required: true,
+          type: InputType.Text
+        },
+        {
+          fieldName: 'description',
+          group: 'main',
+          label: labels?.entity?.description,
+          text: {
+            multilineRows: 3
+          },
+          type: InputType.Text
+        }
+      ],
+      submit: (values, bag) => onSubmit?.(values, bag),
+      validationSchema: Yup.object().shape({
+        description: Yup.string()
+          .label(labels?.entity?.description)
+          .max(
+            180,
+            (p) =>
+              `${p.label} ${t(labelMustBeMost)} ${p.max} ${labelCharacters}`
```

Comment:
I'd create a higher order function that you can reuse the the other validations below this one

Reply:
ok, that's an option to avoid some duplication

better option would be to use `i18n` interpolation (which also supports plurals), because there's no guarantee where `label` and `max` are in a sentence in other languages, and would basically do the same as a custom HoC


```
const minCharTranslationString = "{{label}} should be at least {{minValue}} characters";
const maxCharTranslationString = "{{label}} can be at most {{maxValue}} characters long";

t(minCharTranslationString, { label, minValue })
t(maxCharTranslationString, { label, maxValue })
```




_note: there's a discussion to have on the general approach of `i18n` usage, but considered out-of-scope of this PR_